C4/Items.pm

```diff
@@ -1531,6 +1532,71 @@ sub Item2Marc {
 	return $itemmarc;
 }
 
+=head2 GetMarcItemFields
+
+  my @marc_fields = GetMarcItemFields($biblionumber);
+
+Returns an array of MARC::Record objects of the items for the biblio.
+
+=cut
+
+sub GetMarcItemFields {
+	my ( $biblionumber, $itemnumbers, $hidingrules ) = @_;
+
+    my $item_level_itype = C4::Context->preference('item-level_itypes');
+    # This is so much faster than using Koha::Items->search that it makes sense even if it's ugly.
+    state $sth = C4::Context->dbh->prepare( 'SELECT * FROM items WHERE biblionumber = ?' );
+    $sth->execute( $biblionumber );
+    my $items = $sth->fetchall_arrayref({});
+    $sth->finish();
+    my @item_fields;
+    my ( $itemtag, $itemsubfield ) = GetMarcFromKohaField( 'items.itemnumber' );
+
+    ITEMLOOP: foreach my $item (@$items) {
+
+        # Check itemnumbers
+        next if ( @$itemnumbers && !any { $_ == $item->itemnumber } @$itemnumbers );
```

Comment:
$item is an unblessed reference, right? So $item->itemnumber => $item->{itemnumber}

Reply:
Yeah, now fixed. Thanks for catching that!